### DISCUSSION TITLE
The child-references of a record are not marked as deleted if enableCascadingDelete is TRUE

### DISCUSSION THREAD

**Discussion by Moongazer** [sourcebroker/t3api#26] [Q&A] [answered]
Lets say we have the following record relation: a book (parent) has multiple chapters (children), defined like this:

_TCA configuration:_
```
//book.php
      'ctrl' => [
        'delete' => 'deleted',
      ],
      'columns' => [
        'chapters' => [
            'config' => [
                'type' => 'inline',
                'foreign_table' => 'chapter',
                'foreign_field' => 'book_uid',
                'maxitems' => 9999,
                'behaviour' => [
                    'enableCascadingDelete' => TRUE,
                ],
            ],
        ],
      ],

//chapter.php
      'ctrl' => [
        'delete' => 'deleted',
      ],
```

_Model: Book.php_
```
/**
 * @T3api\ApiResource(
 *     itemOperations={
 *          "delete"={
 *              "method"="DELETE",
 *              "path"="/book/{id}",
 *          },
 *     },
 * )
 */
class Book extends AbstractEntity {
    /**
     * @var ObjectStorage<Chapter>
     * @T3api\Serializer\Groups({
     *     "api_book_post_collection",
     * })
     * @T3api\ORM\Cascade("persist")
     */
    protected $chapters;
}
```

The request `DELETE /api/book/123` will set `deleted=1` for parent record `123`, but not for the child-records.
Instead, if the record `123` is deleted in the TYPO3 backend list view, also the child-records are marked as deleted.

I tried it also with the replaced annotation `@T3api\ORM\Cascade("delete")` but with the same result.

**Comment by klamparski** [comment, marked as answer]
Extbase uses some things from TCA configuration but does not reflect all the features. As far as I know (can't test it at the moment) this one is not reflected and that's why it works for you only in backend. To achieve cascade remove on Extbase you need to use its annotation:

```
**
 * @TYPO3\CMS\Extbase\Annotation\ORM\Cascade("remove")
 */
public $property;
```

Regarding cascade persistence annotation from t3api - it's just security enhancement of Extbase as Extbase does not support any other cascade than remove, unfortunately. We hope some day we can remove it and relay on pure Extbase also for persistence. [Here you can find more info and example usage of cascade persistence](https://docs.typo3.org/p/sourcebroker/t3api/1.1/en-us/HandlingCascadePersistence/Index.html#cascade-persistence-creating-related-entity-in-single-request).

Please try it with Extbase's cascade remove annotation and let us know if it worked. Thanks in advance.

**Comment by Moongazer** [comment]
Yes indeed, that works. Thank you very much for the explanation and sorry for opening the ticket than. Just was a bit confused :+1:
